### PR TITLE
chore: update to polkadot 0.9.38

### PR DESCRIPTION
**Pull request by ntn-x2**
fixes https://github.com/KILTprotocol/ticket/issues/2464
Fixes https://github.com/KILTprotocol/kilt-node/issues/481 -> see issue description for a complete list of PRs.

## Relevant changes

- https://github.com/paritytech/polkadot/pull/4097 (XCM v3)
- https://github.com/paritytech/polkadot/pull/6490 (root origin can issue grants on behalf of the treasury)
- https://github.com/paritytech/substrate/pull/13214 (we used in the the staking pallet but we never called the `set` function, so we don't need to do anything)
- https://github.com/paritytech/substrate/pull/13216 (removes all calls and intrinsics from the authorship pallet)

Release analysis: https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-38/2122

## Open issues

If any of the runtimes is complied with just the `runtime-benchmarks` features, there is a compilation issue due to some nested dependencies. Linked issue: https://github.com/paritytech/cumulus/issues/2230.

### CODE REVIEW

File: runtimes/spiritnet/src/lib.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_version: 11000,
 	impl_version: 0,
 	apis: RUNTIME_API_VERSIONS,
-	transaction_version: 4,
+	transaction_version: 5,
```
Comment (weichweich) [inline runtimes/spiritnet/src/lib.rs:99]: Should we skip a version so that we are in sync with peregrine?

**Review by weichweich — APPROVED** [review]
LGTM

I guess we are introducing the "Prove Size Weights" in a different PR?

**Comment by ntn-x2** [comment]
@weichweich I created a ticket https://github.com/KILTprotocol/ticket/issues/2486. I think we can go ahead an merge this, since the main reason why I did it was the new XCM support.

**Comment by trusch** [comment]
I checked the code and went through all the PRs since 0.9.37 and can't spot anything that's missing :) LGTM :partying_face:

**Review by trusch — APPROVED** [review]

**Pull request merged**